My Jetpack: Updating Stats card on My Jetpack page

grzegorz-cp opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes  MYJP-149

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Added stats chart data fetching functionality to My Jetpack
* Implemented a hook and data transformation for data management for single-metric visualisation
* Added @automattic/charts package dependency
* Implemented interactive stats chart with clickable metric tabs
* Enhanced count comparison cards with selection states and click handlers
* Added dynamic chart titles based on selected metric
* Updated various CSS styling and card layout - introducing a slim card to match the designs
* Configured BarChart component to match the designs
* Implemented proper accessibility features for the card tabs (ARIA attributes, screen reader text)

<img width="1139" alt="SCR-20250619-mqtj" src="https://github.com/user-attachments/assets/e32f28cd-0d00-4c01-8299-bd1860af68b2" />

<img width="1130" alt="SCR-20250619-mqwg" src="https://github.com/user-attachments/assets/9d656aa6-1fd4-46d6-98bb-4641c7447dfc" />

Known issue:
- card is not clickable and doesn't redirect to the Stats
- left side of the chart not aligned with the cards

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Create a JN site for the branch
* Connect Jetpack
* Go to My Jetpack to view the stats card - visit blog posts to generate traffic

grzegorz-cp committed:
Getting chart data

grzegorz-cp committed:
Updating chart and data transformation

grzegorz-cp committed:
Implement interactive stats chart with clickable metric tabs and single-series visualization

grzegorz-cp committed:
Fixing Stats card tab styling

grzegorz-cp committed:
Fixing Stats card tab CSS and label position

grzegorz-cp committed:
Merge branch 'trunk' into update/myjetpack/myjp-149-onboarding-i4-stats-chart

grzegorz-cp committed:
changelog

grzegorz-cp committed:
Merge branch 'trunk' into update/myjetpack/myjp-149-onboarding-i4-stats-chart

grzegorz-cp committed:
Added new card style for Stats in My Jetpack

grzegorz-cp committed:
Empty data notification

grzegorz-cp committed:
Empty data notification styling

grzegorz-cp committed:
Merge branch 'trunk' into update/myjetpack/myjp-149-onboarding-i4-stats-chart

grzegorz-cp committed:
Adding charts package again

manzoorwanijk commented on projects/packages/my-jetpack/_inc/components/stats-section/count-comparison-card.jsx:
```diff
@@ -46,19 +48,52 @@ const CountComparisonCard = ( {
+		<Card
+			className={ cardClassName }
+			as={ as }
+			onClick={ handleClick }
```
Why not simply pass `onClick` here instead of wrapping it in `useCallback`

grzegorz-cp replied:
Sorry for missing this comment earlier. I think that this way is more readable.

manzoorwanijk commented on projects/packages/my-jetpack/_inc/components/stats-section/style.module.scss:
```diff
@@ -26,7 +44,67 @@ $actions-size: 28px;
+		align-self: center;
+		color: var(--color-neutral-80);
+		flex-shrink: 0;
+		margin-right: 20px;
```
RTL aware alternative is better
```suggestion
		margin-inline-end: 20px;
```

manzoorwanijk reviewed (commented):
This works well. The only thing I see different from the design is that there is some margin between the chart border and the measurement horizontal lines.

| Designs | This PR |
|--------|--------|
| <img width="536" alt="Screenshot 2025-06-19 at 10 38 45 AM" src="https://github.com/user-attachments/assets/6c849775-f4a4-49f2-b29b-6ebaaf170a11" /> | <img width="1139" alt="456806657-e32f28cd-0d00-4c01-8299-bd1860af68b2" src="https://github.com/user-attachments/assets/11d0f146-b8d9-4888-a331-06af3c6d737c" /> |

grbicsanja commented:
Thank you, @grzegorz-cp. This looks nice! I caught some inconsistencies between the widget and the chart we use on the stats product page. I'll list them below.

**Design QA**

- The chart area and heading should be clickable and take the user to the product page
- The whole card should be responsive (same behaviour as product page)
- Values for Views, clicks etc are not showing up in numbers, only on the chart
- Last 7 days dates don't match as if they are calculated differently
- Z axis doesn't scale in the same way (see image)
- Chart lines should go all the way to the edge
- Bottom spacing should be 24
- Primary / theme color should be used
- Cards value font size is too big (use product page as reference)
- 0 (0%) for delta is redundant when value is 0

<img width="4789" alt="Frame 2147226152" src="https://github.com/user-attachments/assets/1585a472-91bd-4266-9be1-1d249020ffe7" />
<img width="4789" alt="Frame 2147226153" src="https://github.com/user-attachments/assets/90e26fc6-cc59-49cc-b4a6-87f378516855" />
<img width="4789" alt="Frame 2147226154" src="https://github.com/user-attachments/assets/2d43a8d9-96aa-4079-8eb6-2438c5916ad8" />

grzegorz-cp committed:
Updating CSS after review, adding click redirect

grzegorz-cp committed:
Merge branch 'trunk' into update/myjetpack/myjp-149-onboarding-i4-stats-chart

grzegorz-cp commented:
Thank you for the reviews @grbicsanja and @manzoorwanijk. I added click functionality and updated CSS to match the other cards.
Let me know what you think. Since this PR is getting quite big I'd like to ship it if you think we can do it and add make the remaining changes in another one.

@grbicsanja thanks for catching the difference between cards and chart. I was using two endpoints that didn't return the same data but I fixed it.

<img width="1068" alt="SCR-20250620-njtv" src="https://github.com/user-attachments/assets/78c4c129-ca2c-4371-aef3-7de079716803" />

grzegorz-cp committed:
pnpm lock update

grzegorz-cp committed:
script changes

manzoorwanijk reviewed (approved):
Yeah, this is getting big. Let us make other improvements in a follow up. But I have a concern that this is going to trunk instead of the feature branch. Should we keep the other changes ready before merging this?

gmjuhasz committed:
Social Previews: Fix Mastodon preview stretching (#44019)

* Fix Mastodon description overflow with previews

* changelog

* Lockfile

* Bump to 12

mmtr committed:
Admin Menu: Add "Dashboard > Updates" screen to Simple sites (#44024)

* Admin Menu: Add "Dashboard > Updates" menu to Simple sites

* Remove WP version

* Add changelog

* Exclude VIP sites, since the plugins (and their updates) are managed a little bit differently there, and parity with Atomic sites is also not important for them.

* Check that function exists

heavyweight committed:
Zendesk: Add CSAT endpoint (#44029)

jeherve committed:
Blocks: In scan_delimiters extract static method (#43984)

Ports 185482-ghe-Automattic/wpcom

When `scan_delimiters()` is attempting to find the end of an HTML
comment, it’s calling a convenience method which is also a closure
in the code (or an anonymous function). This carries a marginal
performance overhead and limits reuse.

In this change that method is extracted onto the class as its own static
method; this should simplify reasoning about the method itself and
provide a perofmrnace lift to the calling code.

Co-authored-by: Dennis Snell <dennis.snell@automattic.com>
Co-authored-by: Brandon Kraft <public@brandonkraft.com>

tbradsha committed:
WordAds: Remove package from monorepo (#44025)

* Fix test description

* Delete packages/wordads

* Add changelog

* Fix project lint issues

jeherve committed:
Subscriptions: fix subcriber stats links when stats module is not active (#44037)

* Subscriber dashboard: pass stats module status to JS

We will use it to change some of the links when the module is not active.

* Add extra space for better readability

* Pass new prop to newsletter component

* Use new prop to change the status of the links

- On WordPress.com sites, always link to WordPress.com.
- On self-hosted sites, check for the status of the Stats module:
    * If it's active, link to the wp-admin stats page.
    * If it is not active, link to the Jetpack cloud subscribers page.

* Update test accordingly

* Update dashboard links to be aware of stats module status

* Add changelog

* Redirect folks to Calypso blue stats when stats module is inactive

This is now possible because the Subscribers stats sub-page of the Calypso blue stats is not accessible, even when you do not use the Stats module.

See pejTkB-24B-p2

* simplify internal link logic

See https://github.com/Automattic/jetpack/pull/44037/files#r2158706216

* Update doc comment

* Be more explicit about stats being enabled in original test

See https://github.com/Automattic/jetpack/pull/44037#discussion_r2159060667

Co-authored-by: Allison Levine <1689238+allilevine@users.noreply.github.com>

---------

Co-authored-by: Tony Arcangelini <tony@arcangelini.com>
Co-authored-by: Allison Levine <1689238+allilevine@users.noreply.github.com>

matticbot committed:
Update dependency copy-webpack-plugin to v13 (#44040)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

grzegorz-cp committed:
fixing package dependency

grzegorz-cp committed:
Update margin-inline-end

Co-authored-by: Manzoor Wani <manzoorwani.jk@gmail.com>

grzegorz-cp committed:
Revert "fixing package dependency"

This reverts commit b592fd97eaa7c2614b4ab3c20747f77f70ad4d65.

grzegorz-cp committed:
Merge branch 'trunk' into update/myjetpack/myjp-149-onboarding-i4-stats-chart

grzegorz-cp committed:
Charts webpack fix

grzegorz-cp committed:
changelog

grzegorz-cp committed:
Fixing timezones to match Stats and hiding 0 diff values for catds

keoshi reviewed (commented):
Overall a few styles are off between the new widget and the Stats page. I understand we’re creating the widget from scratch so take these with a grain of salt and not as blockers.

This also seems like a huge push, so I'm approving this PR with the intention that we can address smaller issues separately.

> Let's align the lines to the left with the boxes

- Would even say to replicate the look of the current chart module in Stats, meaning pushing the lines to the edges.

Stats | My JP
--|--
<img width="212" alt="image" src="https://github.com/user-attachments/assets/360e7991-aeb4-4bcf-9486-8a0ba0a5a67e" /> | <img width="189" alt="image" src="https://github.com/user-attachments/assets/5b00bc3a-3a5f-4dce-a554-5a38d1723d10" />

- The padding should be the same around the entire widget box.
- Also notice how the selected style doesn't match the Stats page. Border width should be `1.5px`.
- Finally, revise the vertical spacing between the icon+label and the number below.

Stats | My JP
--|--
<img width="314" alt="image" src="https://github.com/user-attachments/assets/af84ebbb-a024-47f5-a039-21ecea8fd7bd" /> | <img width="289" alt="image" src="https://github.com/user-attachments/assets/e05d9864-63e1-4e7a-b60c-176525d7ed42" />

- Gap between icon and label should be `4px`.

Stats | My JP
--|--
<img width="110" alt="image" src="https://github.com/user-attachments/assets/287a3b5e-78e0-4a25-929b-326464168e7e" /> | <img width="105" alt="image" src="https://github.com/user-attachments/assets/5bdb6ad7-d5f3-4d0d-add6-e2d143b7c6aa" />
